ui/packages/components/src/TimelineV2/types.ts

```diff
@@ -9,6 +9,51 @@ export type Trace = {
   spanID: string;
   startedAt: string | null;
   status: string;
-  stepInfo?: unknown;
+  stepInfo: StepInfoInvoke | StepInfoSleep | StepInfoWait | null;
   stepOp?: string | null;
 };
+
+type StepInfoInvoke = {
+  triggeringEventID: string;
+  functionID: string;
+  timeout: string;
+  returnEventID: string | null;
+  runID: string | null;
+  timedOut: boolean;
+};
+
+type StepInfoSleep = {
+  sleepUntil: string;
+};
+
+type StepInfoWait = {
+  eventName: string;
+  expression: string | null;
+  timeout: string;
+  foundEventID: string | null;
+  timedOut: boolean;
+};
+
+export function isStepInfoInvoke(stepInfo: Trace['stepInfo']): stepInfo is StepInfoInvoke {
```

Comment:
seems to me these three function could be consolidated to a single function if they took an additional param of 'triggeringEventID' | 'sleepUntil' | 'foundEventID'.